src/schema-association-service-1espt.ts

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License.
+*--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Utils } from 'vscode-uri';
+import * as azdev from 'azure-devops-node-api';
+import * as logger from './logger';
+import { AzureSession } from './typings/azure-account.api';
+import { VersionControlRecursionType } from 'azure-devops-node-api/interfaces/TfvcInterfaces';
+import * as fs from 'fs'
+import { Messages } from './messages';
+
+const milliseconds24hours = 86400000;
+
+export async function get1ESPTSchemaUriIfAvailable(azureDevOpsClient: azdev.WebApi, organizationName: string, session: AzureSession, context: vscode.ExtensionContext, lastUpdated1ESPTSchema: Date, seen1ESPTOrganizations: Set<string>): Promise<any> {
+    try {
+        if (session.userId.endsWith("@microsoft.com")) {
+            const gitApi = await azureDevOpsClient.getGitApi();
+            const repositories = await gitApi.getRepositories('1ESPipelineTemplates');
+            if (!repositories || repositories.length == 0) {
+                logger.log(`1ESPT repo not found for org ${organizationName}`, `SchemaDetection`)
+                const config = vscode.workspace.getConfiguration('azure-pipelines')
+                config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.Global); // disable the 1ESPT schema configuration
+                vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsADOOrgNotContains1ESPT)
+                return undefined; // 1ESPT repo not found
+            }
+            const repository = repositories.find(repo => repo.name === "1ESPipelineTemplates");
+            var schemaFile = await gitApi.getItem(repository.id, "schema/1espt-base-schema.json", "1ESPipelineTemplates", "", VersionControlRecursionType.None, true, true, true, {}, true, true)
+
+            const schemaContent = schemaFile.content
+            const schemaUri = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+            await vscode.workspace.fs.writeFile(schemaUri, Buffer.from(schemaContent));
+            lastUpdated1ESPTSchema = new Date();
+            seen1ESPTOrganizations.add(organizationName);
+            return schemaUri
+        }
+        else
+        // if user is signed in with account other than microsoft, then disable 1ESPT schema and delete the 1ESPT schema file
```

Comment:
Same concern - do we need to disable the setting? If they're not Microsoft then the fallback can be silent so they don't need to remember to re-enable if they ever do log in with their Microsoft account.

Reply:
We discussed this internally. Since sign out and sign back in is a rare scenario, we decided to keep it this way. User will see a message window asking them to enable it when they sign in with Microsoft account